# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/style/nsStyleTransformMatrix.h
# Commit: d9b057905efd
# Full Hash: d9b057905efd5a2feedf354ea5b8a039a042ae09
# Author: Boris Chiou <boris.chiou@gmail.com>
# Date: 2023-08-23 04:21:33
# Regressor Bug: 1819464
# File Overlap Count: 1
# Description:
#   Bug 1819464 - Part 3: Refactor TransformReferenceBox. r=emilio
#   
#   Refactor this structure to match spec and make us easier to support
#   content-box and stroke-box.
#   
# ==============================================================================

diff -r 473ebe5ec7aa -r d9b057905efd layout/style/nsStyleTransformMatrix.h
--- a/layout/style/nsStyleTransformMatrix.h	Tue Aug 22 19:49:03 2023 +0000
+++ b/layout/style/nsStyleTransformMatrix.h	Tue Aug 22 19:49:03 2023 +0000
@@ -61,24 +61,15 @@
  public:
   typedef nscoord (TransformReferenceBox::*DimensionGetter)();
 
-  explicit TransformReferenceBox()
-      : mFrame(nullptr),
-        mX(0),
-        mY(0),
-        mWidth(0),
-        mHeight(0),
-        mIsCached(false) {}
+  TransformReferenceBox() = default;
 
-  explicit TransformReferenceBox(const nsIFrame* aFrame)
-      : mFrame(aFrame), mX(0), mY(0), mWidth(0), mHeight(0), mIsCached(false) {
+  explicit TransformReferenceBox(const nsIFrame* aFrame) : mFrame(aFrame) {
     MOZ_ASSERT(mFrame);
   }
 
-  explicit TransformReferenceBox(const nsIFrame* aFrame,
-                                 const nsRect& aFallbackDimensions)
-      : mX(0), mY(0), mWidth(0), mHeight(0) {
+  TransformReferenceBox(const nsIFrame* aFrame,
+                        const nsRect& aFallbackDimensions) {
     mFrame = aFrame;
-    mIsCached = false;
     if (!mFrame) {
       Init(aFallbackDimensions);
     }
@@ -89,7 +80,11 @@
     mFrame = aFrame;
   }
 
-  void Init(const nsRect& aDimensions);
+  void Init(const nsRect& aDimensions) {
+    MOZ_ASSERT(!mFrame && !mIsCached);
+    mBox = aDimensions;
+    mIsCached = true;
+  }
 
   /**
    * The offset of the reference box from the nsIFrame's TopLeft(). This
@@ -99,11 +94,11 @@
    */
   nscoord X() {
     EnsureDimensionsAreCached();
-    return mX;
+    return mBox.X();
   }
   nscoord Y() {
     EnsureDimensionsAreCached();
-    return mY;
+    return mBox.Y();
   }
 
   /**
@@ -111,11 +106,11 @@
    */
   nscoord Width() {
     EnsureDimensionsAreCached();
-    return mWidth;
+    return mBox.Width();
   }
   nscoord Height() {
     EnsureDimensionsAreCached();
-    return mHeight;
+    return mBox.Height();
   }
 
   bool IsEmpty() { return !mFrame; }
@@ -128,9 +123,9 @@
 
   void EnsureDimensionsAreCached();
 
-  const nsIFrame* mFrame;
-  nscoord mX, mY, mWidth, mHeight;
-  bool mIsCached;
+  const nsIFrame* mFrame = nullptr;
+  nsRect mBox;
+  bool mIsCached = false;
 };
 
 float ProcessTranslatePart(
